File: fm-dddrive/src/main/java/io/dddrive/core/ddd/model/PartSPI.java
package io.dddrive.core.ddd.model;

/**
 * This interface defines the internal callbacks for a Part implementation.
 */
public interface PartSPI<A extends Aggregate> {

	/**
	 * Initialise a Part after creation (external, functional callback).
	 */
	default void doAfterCreate() {}

	/**
	 * Initialise a Part after load (external, functional callback).
	 */
	void doAfterLoad();

	void delete();

}

File: fm-dddrive/src/main/java/dddrive/ddd/model/PartSPI.kt
package dddrive.ddd.model

/**
 * This interface defines the internal callbacks for a Part implementation.
 */
interface PartSPI<A : Aggregate> {

	/**
	 * Initialise a Part after creation (external, functional callback).
	 */
	fun doAfterCreate() {}

	fun delete()

}
